pallets/protos/src/lib.rs

```diff
@@ -809,25 +831,46 @@ pub mod pallet {
 						}
 					}
 
-					let cost = proto.include_cost;
-					if let Some(cost) = cost {
-						let cost: u64 = cost.into();
-						if cost == 0 {
-							// free
-							continue;
-						}
-						let cost: T::Balance = cost.saturated_into();
-						let stake = <ProtoStakes<T>>::get(reference, who.clone());
-						if let Some(stake) = stake {
-							ensure!(stake.0 >= cost, Error::<T>::NotEnoughStaked);
-						} else {
-							// Stake not found
-							return Err(Error::<T>::StakeNotFound.into());
+					let license = proto.license;
+					match license {
+						UsageLicense::NotAvailable => {
+							return Err(Error::<T>::Unauthorized.into());
+						},
+						UsageLicense::Available => { continue },
+						UsageLicense::Tickets(amount) => {
+							let balance = <pallet_assets::Pallet<T> as Inspect<T::AccountId>>::balance(T::TicketsAssetId::get(), &who);
+							let amount: u64 = amount.into();
+							ensure!(balance >= amount.saturated_into(), Error::<T>::InsufficientBalance);
+							// TODO transfer tickets
```

Comment:
What do you mean with _transfer tickets_?

Reply:
wip, was wrong